### PR TITLE
Simplified code for edge features

### LINKED ISSUE #207: Explore `add_features_for_residues` function in feature/atomic_contact.py
Currently nested for loops are used to compare each atom of one residue against each atom of the other residue and extract information one comparison at a time.
There is likely a way to get the sum/minimum/etc (whichever is applicable) in one shot, perhaps using some numpy function. 

This is likely to be faster and definitely more pythonic and readable.

### PR DESCRIPTION
**Pull request by DaniBodor**
Massively simplified code for addition of edge features in features.atomic_contact.
Also fixed mistake in LJ potential calculation
Removed old functions from tools.pdb, which I think were intended for or once upon a time used for generating edge features, but is now done in feature.atomic_contact

closes #207 and #218

### CODE REVIEW

File: deeprankcore/domain/forcefield/__init__.py [hunk 1]
```diff
@@ -58,17 +59,12 @@ def _find_matching_residue_class(self, residue: Residue):
...
-    def _get_type(self, atom: Atom):
         atom_name = atom.name
 
         if atom.residue.amino_acid is None:
-            raise UnknownAtomError(f"no amino acid for {atom}")
-
-        residue_name = atom.residue.amino_acid.three_letter_code
+            _log.warning(f"no amino acid for {atom}; three letter code set to XXX")
+            residue_name = 'XXX'
+        else: residue_name = atom.residue.amino_acid.three_letter_code
```
Comment (DaniBodor) [inline deeprankcore/domain/forcefield/__init__.py:68]: this error (and below) used to be caught and handled in the code calling this function. I just substituted that warning into here instead.

File: deeprankcore/domain/forcefield/__init__.py [hunk 2]
```diff
@@ -58,17 +59,12 @@ def _find_matching_residue_class(self, residue: Residue):
...
-
-        return self._vanderwaals_parameters[type_]
-
-    def _get_type(self, atom: Atom):
         atom_name = atom.name
 
         if atom.residue.amino_acid is None:
-            raise UnknownAtomError(f"no amino acid for {atom}")
-
-        residue_name = atom.residue.amino_acid.three_letter_code
```
Comment (DaniBodor) [inline deeprankcore/domain/forcefield/__init__.py:71]: this and below UnknownAtomErrors used to be caught in the code that calls the functions. I just substituted the warnings raised there in here.

**Comment by DaniBodor** [comment]
All tests run fine locally, not sure why they fail here

File: deeprankcore/domain/forcefield/__init__.py [hunk 3]
```diff
@@ -58,17 +59,12 @@ def _find_matching_residue_class(self, residue: Residue):
         return None
 
     def get_vanderwaals_parameters(self, atom: Atom):
-        type_ = self._get_type(atom)
```
Comment (gcroci2) [inline deeprankcore/domain/forcefield/__init__.py:61]: Don't we need type_ var?

**Review by gcroci2 — APPROVED** [review]

File: deeprankcore/domain/forcefield/__init__.py (hunk 3 above)
Comment (cbaakman) [reply deeprankcore/domain/forcefield/__init__.py:61]: It's used for getting the right vanderwaals parameters from the table file.

**Review by gcroci2 — COMMENTED** [review]
Seems good, but the build is failing. Please fix it before merging :)

**Comment by DaniBodor** [comment]
I noticed that this code is actually slower than the original in assigning features. I tested this by looping some tests from test_query and checking how long it takes to run.

I realize that this is (probably) because before that calculations (e.g. `get_coulomb_potentials` and `get_lennard_jones_potentials`) were only run once per graph on many atoms, and now they are run once per edge on few atoms at a time. Apparently the new way is slower.

I can look into finding a way to combine what happened before with the simplified code base, or we can revert to how it was or leave it as is. What do you think?

**Comment by gcroci2** [comment]
> I noticed that this code is actually slower than the original in assigning features. I tested this by looping some tests from test_query and checking how long it takes to run.
> 
> I realize that this is (probably) because before that calculations (e.g. `get_coulomb_potentials` and `get_lennard_jones_potentials`) were only run once per graph on many atoms, and now they are run once per edge on few atoms at a time. Apparently the new way is slower.
> 
> I can look into finding a way to combine what happened before with the simplified code base, or we can revert to how it was or leave it as is. What do you think?

I would just revert it. It would have been a nice addition but I wouldn't lose too much time on this

File: deeprankcore/domain/forcefield/__init__.py (hunk 3 above)
Comment (DaniBodor) [reply deeprankcore/domain/forcefield/__init__.py:61]: As far as I could see it was only used internally by the `get_vanderwaals_parameters` function. However, this function basically did nothing else than read the type_ and get the parameters, so instead I merged `get_type` into `get_vanderwaals_parameters`.
I tested that old and new `add_features` functions (from atomic_contact, which uses this) give the exact same result on ~5 test cases, so am fairly confident that it is still being read correctly.

**Comment by DaniBodor** [comment]
> I would just revert it. It would have been a nice addition but I wouldn't lose too much time on this

I resolved it. Didn't take that long and wanted to use some of the fixes from the PR anyway.

**Review by gcroci2 — APPROVED** [review]

**Pull request merged**